### DISCUSSION TITLE
Example class with Freezed with multiple constructors

### DISCUSSION THREAD

**Discussion by jy95** [isar/isar#191] [Help] [answered]
Hello,

I wonder how to make Isar works with multiple constructors in Freezed such as :

```dart
@freezed
class Example with _$Example {
  const factory Example.person(String name, int age) = Person;
  const factory Example.city(String name, int population) = City;
}
```

Thanks in advance

**Comment by simc** [comment, marked as answer]
This won't work since it basically generates two classes (person and city). What would you Isar expect to do?

**Reply by jy95** [reply]
I might be wrong but something like that :

```dart
 @freezed
class Example with _$Example {
  @Collection()
  const factory Example.person({
     @Id() int? id,
     String name, 
     int age
  }) = Person;
  @Collection()
  const factory Example.city({
     @Id() int? id,
     String name, 
     int population
  }) = City;
}
```

If you want a realistic example, consider a Recipe class that holds a list of ingredients (RecipeIngredient class) that are either freeText (NotCodifiedIngredient class) or codified (CodifiedIngredient class).

**Reply by simc** [reply]
Yes, but Isar needs a fixed schema. It can deal with a person collection and a city collection. But there is no such thing as a "PersonOrCity" collection.

**Reply by jy95** [reply]
That is what I thought.
Time to lookup for a local flutter database that allows me to query JSON like in Postgresql ;)
